compiler/rustc_mir_dataflow/src/framework/engine.rs

```diff
@@ -146,12 +146,19 @@ where
         analysis: A,
         apply_trans_for_block: Option<Box<dyn Fn(BasicBlock, &mut A::Domain)>>,
     ) -> Self {
-        let bottom_value = analysis.bottom_value(body);
-        let mut entry_sets = IndexVec::from_elem(bottom_value.clone(), &body.basic_blocks);
-        analysis.initialize_start_block(body, &mut entry_sets[mir::START_BLOCK]);
+        let mut init_value = analysis.bottom_value(body);
+        let mut entry_sets = IndexVec::from_elem(init_value.clone(), &body.basic_blocks);
 
-        if A::Direction::IS_BACKWARD && entry_sets[mir::START_BLOCK] != bottom_value {
-            bug!("`initialize_start_block` is not yet supported for backward dataflow analyses");
+        if A::Direction::IS_FORWARD {
+            analysis.initialize_start_block(body, &mut entry_sets[mir::START_BLOCK]);
+        } else {
+            // For backward analyses, initialize blocks that exit MIR.
+            analysis.initialize_start_block(body, &mut init_value);
```

Comment:
Using `initialize_start_block` for this feels weird. Can you instead treat locals with debuginfo as being read on return/resume terminators?